lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -79,17 +80,95 @@ static void emitDxilDiag(LLVMContext &Ctx, const char *str) {
   hlsl::dxilutil::EmitErrorOnContext(Ctx, str);
 }
 
+class StringTableVerifier {
+  std::unordered_map<unsigned, unsigned> OffsetToUseCountMap;
+  const PSVStringTable &Table;
+
+public:
+  StringTableVerifier(const PSVStringTable &Table) : Table(Table) {
+    unsigned Start = 0;
+    for (unsigned i = 0; i < Table.Size; ++i) {
+      char ch = Table.Table[i];
+      if (ch == '\0') {
+        OffsetToUseCountMap[Start] = 0;
+        Start = i + 1;
+      }
+    }
+    if (Table.Size >= 4) {
+      // Remove the '\0's at the end of the table added for padding.
+      for (unsigned i = Table.Size - 1; i > Table.Size - 4; --i) {
+        if (Table.Table[i] != '\0')
+          break;
+        OffsetToUseCountMap.erase(i);
```

Comment:
Won't this potentially attempt to erase a non-existent entry pointing to the null terminating character of the last string in the table?

Reply:
I think my concern was this:
Say the string table is: "12\0\0"
There will be an `OffsetToUseCountMap` entry pointing at the last '\0', but not the second to last one that is the null-termination character for the string "12".  This loop will still iterate to this character though, and `Table.Table[i] == '\0'`, so it will attempt to erase an entry from `OffsetToUseCountMap` that doesn't exist.